include/sg14/bits/fixed_point_math.h

```diff
@@ -0,0 +1,126 @@
+//          Copyright Timo Alho 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file ../LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file
+/// \brief some cmath specializations for `sg14::fixed_point` type;
+
+#ifndef FIXED_POINT_MATH_H_
+#define FIXED_POINT_MATH_H_
+
+#include <sg14/fixed_point.h>
+#include <type_traits>
+
+namespace sg14 {
+
+namespace _fixed_point_impl {
+
+template<typename fp>
```

Comment:
A few style points:
- `class` is preferred to `typename` in template parameter lists. (This is a legacy thing mentioned in the [Standard Library Guidelines](https://isocpp.org/std/library-design-guidelines).)
- Template parameters are in CamelCase.
- I try and avoid the acronym F.P. as if can refer to fixed-point but also floating point.